common/src/main/java/org/conscrypt/OpenSSLAeadCipher.java

```diff
@@ -220,6 +220,52 @@ boolean allowsNonceReuse() {
         return false;
     }
 
+    /**
+     * Encrypts of decrypts data in a single-part operations or finishes multiple part operation.
+     * @param input Input data that is returned fully read
+     * @param output the storage buffer of the resulting encryption decryption. Output is returned position
+     *              incremented by bytesWritten Limit stays the same
+     * @return number of bytes the output buffer's position has moved by.
+     * @throws ShortBufferException if output.remaining() bytes are insufficient to hold the result.
+     * @throws IllegalBlockSizeException
+     * @throws BadPaddingException
+     */
+    @Override
+    protected int engineDoFinal(ByteBuffer input, ByteBuffer output) throws ShortBufferException, IllegalBlockSizeException, BadPaddingException {
+        if (input == null || output == null) {
+            throw new NullPointerException("Null ByteBuffer Error");
+        }
+        if ( getOutputSizeForFinal(input.remaining()) > output.remaining()) { // uses the child class's size if available
+            throw new ShortBufferException("Insufficient Bytes for Output Buffer");
```

Comment:
throw `ShortBufferWithoutStackTraceException`, please.  See #849

Reply:
refreshed from the upstream and added this thank you!